internal/application/services.go

```diff
@@ -15,6 +17,173 @@ import (
 
 type NameSet map[string]struct{}
 
+// BoundApps is an extension of BoundAppsNames after it, to retrieve a map of services to
+// the full data of the applications bound to them. It uses BoundAppsNames internally to
+// quickly determine the applications to fetch.
+
+func BoundApps(ctx context.Context, cluster *kubernetes.Cluster, namespace string) (map[string]models.AppList, error) {
+
+	result := map[string]models.AppList{}
+
+	bindings, err := BoundAppsNames(ctx, cluster, namespace)
+	if err != nil {
+		return result, err
+	}
+
+	// Internal map of fetched applications.
+	fetched := map[string]*models.App{}
+
+	if namespace == "" {
```

Comment:
this if/else won't be needed if with the suggested refactoring here: https://github.com/epinio/epinio/pull/1014/files#r756052176

Reply:
commit 858aeddc147ade02f2e1c91012b1809ac3f6989a